Sources/PrimerSDK/Classes/Core/Analytics/AnalyticsService.swift

```diff
@@ -0,0 +1,220 @@
+//
+//  Analytics.swift
+//  PrimerSDK
+//
+//  Created by Evangelos on 13/12/21.
+//
+
+#if canImport(UIKit)
+
+import Foundation
+
+extension Analytics {
+    
+    internal class Service {
```

Comment:
could we make this protocol-based? There are some places in the code (utils, extensions, etc.) where we reference this service, but it feels like it would be better to inject it as a protocol so that it's easier to write unit tests, etc.

Reply:
@ceriksson yes, and this one makes sense to be added in the dependency container.